server/app/auth/CiviFormProfile.java

```diff
@@ -229,8 +229,9 @@ public CompletableFuture<Void> checkProgramAuthorization(String programName) {
     return this.getAccount()
         .thenApply(
             account -> {
-              if (account.getAdministeredProgramNames().stream()
-                  .anyMatch(program -> program.equals(programName))) {
+              if (account.getGlobalAdmin()
+                  || account.getAdministeredProgramNames().stream()
+                      .anyMatch(program -> program.equals(programName))) {
```

Comment:
issue (blocking): global admins are not guaranteed to be admins of a specific program, let's talk about alternatives for addressing the issue